docs/syntaxes.md

```diff
@@ -0,0 +1,221 @@
+between them by using the "Syntax" widget in the top left corner of this page.
+Try clicking on it now to see the two different versions of the following code
+snippet:
+
+```ocaml
+module Speech = struct
+  type utterance
+
+  external makeUtterance : string -> utterance = "SpeechSynthesisUtterance"
+  [@@mel.new]
+
+  external speak : utterance -> unit = "speechSynthesis.speak"
+end
+
+module App = struct
+  let style =
+    ReactDOM.Style.make ~fontSize:"1em" ~border:"1px solid white"
+      ~borderRadius:"0.5em" ~padding:"1em" ()
+
+  let make () =
+    (button ~style
+       ~onClick:(fun _ ->
+         "Hello ReasonReact" |> Speech.makeUtterance |> Speech.speak)
+       ~children:[ React.string "Say hello" ]
+       () [@JSX])
+  [@@react.component]
+end
+
+let () =
+  match ReactDOM.querySelector "#preview" with
+  | None -> Js.log "Failed to start React: couldn't find the #preview element"
+  | Some root ->
+      let root = ReactDOM.Client.createRoot root in
+      ReactDOM.Client.render root (App.createElement ~children:[] () [@JSX])
+```
+```reasonml
+module Speech = {
+  type utterance;
+
+  [@mel.new]
+  external makeUtterance: string => utterance = "SpeechSynthesisUtterance";
+
+  external speak: utterance => unit = "speechSynthesis.speak";
+};
+
+module App = {
+  let style =
+    ReactDOM.Style.make(
+      ~fontSize="1em",
+      ~border="1px solid white",
+      ~borderRadius="0.5em",
+      ~padding="1em",
+      (),
+    );
+
+  [@react.component]
+  let make = () =>
+    <button
+      style
+      onClick={_ =>
+        "Hello ReasonReact" |> Speech.makeUtterance |> Speech.speak
+      }>
+      {React.string("Say hello")}
+    </button>;
+};
+
+let () =
+  switch (ReactDOM.querySelector("#preview")) {
+  | None =>
+    Js.log("Failed to start React: couldn't find the #preview element")
+  | Some(root) =>
+    let root = ReactDOM.Client.createRoot(root);
+    ReactDOM.Client.render(root, <App />);
+  };
+```
+
+As you can see, Reason syntax resembles JavaScript and supports JSX, so it is
+more approachable to JavaScript and TypeScript developers. Most tooling, such as
+the [OCaml Platform VS Code
+extension](https://marketplace.visualstudio.com/items?itemName=ocamllabs.ocaml-platform),
+works well with Reason, but some tools aren't completely aware of it—for
+example, error messages from the compiler still use OCaml syntax.
+
+## Installation
+
+### Reason syntax
+
+To install Reason syntax support add `reason` to the `depends` section of the
+`<project-name>.opam` file in your project's root directory:
+
+```opam
+depends: [
+  "ocaml"
+  "reason" {>= "3.10.0"} # [!code ++]
+  "dune" {>= "3.9"}
+  "melange" {>= "2.0.0"}
+  "reason-react" {>= "0.13.0"}
+  "reason-react-ppx"
+  "opam-check-npm-deps" {with-dev-setup}
+  "ocaml-lsp-server" {with-dev-setup}
+  "dot-merlin-reader" {with-setup}
+  "odoc" {with-doc}
+]
+```
+
+On the command line, run
+
+```bash
+opam install -y . --deps-only
```

Comment:
Should we mention that alternatively, users can just run `opam install reason -y`?

Reply:
Added alternate command in https://github.com/melange-re/melange-re.github.io/pull/200/commits/55b49d484671e0cd98febae5deaeaa332c99a785